src/app/services/polling.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+
+import {
+  BehaviorSubject,
+  delay,
+  from,
+  fromEvent,
+  mergeMap,
+  of,
+  repeatWhen,
+  share,
+  switchMap,
+  takeWhile,
+  timer,
+} from 'rxjs';
+
+import { OnlineStatusService, OnlineStatusType } from 'ngx-online-status';
+import { BnNgIdleService } from 'bn-ng-idle';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PollingService {
+  private renewStream = new BehaviorSubject<boolean>(true);
+
+  constructor(private onlineStatusService: OnlineStatusService, private countIdleService: BnNgIdleService) {}
+
+  fetchUrl = () => of(`Generating http response ${Date.now()}`).pipe(delay(100));
+
+  startPolling(): void {
```

Comment:
better idea of how it can work - we have service with method (like startPolling) which returns an observable which emits value when all our rules satisfied
then, we can use it in any place:
```
this.pollingService.startPolling().pipe(
    takeUntil(this.unsubscribe)
).subscribe(() => {
    this.dateService.getDataWeNeed();
});
```

Reply:
hint - we can have variables to check if tab is active, if user is online inside method itself, we don't need any behaviourSubject etc